biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java

```diff
@@ -594,29 +594,32 @@ private boolean isMaster(InfoRepository repo, String bsn, Version v) throws Exce
 	}
 
 	private RepositoryPlugin getReleaseRepo() {
-		String repoName = getProperty(Constants.RELEASEREPO);
 
-		List<RepositoryPlugin> repos = getPlugins(RepositoryPlugin.class);
-		for (RepositoryPlugin r : repos) {
-			if (r.canWrite()) {
-				if (repoName == null || r.getName()
-					.equals(repoName)) {
-					return r;
-				}
-			}
-		}
-		if (repoName == null)
+		String repoNames = getProperty(Constants.RELEASEREPO);
+
+		if (repoNames == null) {
 			error("Could not find a writable repo for the release repo (-releaserepo is not set)");
-		else
-			error("No such -releaserepo %s found", repoName);
+			return null;
+		}
+
+		List<RepositoryPlugin> releaseRepos = project.getReleaseRepos(repoNames);
 
+		if (!releaseRepos.isEmpty()) {
+			warning("Found multiple release repositories [%s], so we will use the first one", repoNames);
```

Comment:
Not empty can also mean 1. So the warning will trigger when there is only one repo in the list.